subsys/bluetooth/controller/ticker/ticker.c

```diff
@@ -2276,11 +2276,18 @@ void ticker_job(void *param)
 
 	DEBUG_TICKER_JOB(1);
 
-	/* Defer worker, as job is now running */
+	/* Defer job, as worker is running */
 	if (instance->worker_trigger) {
 		DEBUG_TICKER_JOB(0);
 		return;
 	}
+
+	/* Defer job, as job is already running */
+	if (instance->job_guard) {
+		instance->sched_cb(TICKER_CALL_ID_JOB, TICKER_CALL_ID_JOB, 1,
+				   instance);
+		return;
+	}
 	instance->job_guard = 1U;
 
```

Comment:
Consider also moving clearing of the job_guard to after `ticker_job_compare_update`

Reply:
Is this because `ticks_slot_previous` and `ticks_current` are updated in `ticker_job_compare_update`, and the vectoring (swap) to `ticker_worker` function would use stale values? or do you have other reasons? 

(sorry for asking again, from the message here: https://github.com/zephyrproject-rtos/zephyr/pull/31026#issue-546079708, I understand context-safety, but just making double sure :-) ).

Reply:
> Is this because `ticks_slot_previous` and `ticks_current` are updated in `ticker_job_compare_update`, and the vectoring (swap) to `ticker_worker` function would use stale values? or do you have other reasons?

Yes, that is exactly why. The suggestion is not based on an actual observation, just based on analysis and being cautious.

Reply:
Agreed. Your analysis is correct, as the resolution happens in ticker_worker, there are chances that after a current reserving ticker is stopped, it is necessary to have correct value of `ticks_slot_previous` and `ticks_current` be used in the ticker_worker.